### PR TITLE
fixed https://wso2.org/jira/browse/APIMANAGER-4066

### PR DESCRIPTION
**Pull request by ruks**
fixed https://wso2.org/jira/browse/APIMANAGER-4066

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java [hunk 1]
```diff
@@ -2238,7 +2238,15 @@ public static NativeArray jsFunction_getAPI(Context cx, Scriptable thisObj,
                 PrivilegedCarbonContext.startTenantFlow();
                 PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
             }
-            API api = apiProvider.getAPI(apiId);
+
+            API api = null;
+            try {
+                api = apiProvider.getAPI(apiId);
+            } catch (APIManagementException e) {
+                handleException("Cannot find the requested API- " + apiName +
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: handle exception also return APIManagementException which means catch and throw same exception

Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: same exception catch here to send a custom error message indicating that it is occurred due to API is not existing.

Comment (prabathabey) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: Original exception trace is swallowed. Please fix!

Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: Original exception trace is already fixed in commit: https://github.com/wso2/carbon-apimgt/commit/e433e625a5bed755ccab07851ac9510361b9e400

**Pull request merged**